frontend/packages/console-shared/src/hooks/select-list.ts

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import { OnSelect } from '@patternfly/react-table';
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+
+export const useSelectList = <R extends K8sResourceCommon>(
+  data: R[],
+  visibleRows: Set<string>,
+  onRowSelected: (rows: R[]) => void,
+): {
+  onSelect: OnSelect;
+  selectedRows: Set<string>;
+  updateSelectedRows: (rows: R[]) => void;
+} => {
+  const [selectedRows, setSelectedRows] = React.useState<Set<string>>();
```

Comment:
Lets initialize with empty Set
```suggestion
  const [selectedRows, setSelectedRows] = React.useState<Set<string>>(new Set());
```